modules/misc/qt.nix

```diff
@@ -55,56 +56,85 @@ in {
 
-      platformTheme = lib.mkOption {
+      platformTheme = let
+        newOption = {
+          name = lib.mkOption {
+            type = with lib.types; nullOr str;
+            default = null;
+            example = "adwaita";
+            relatedPackages = [
+              "qgnomeplatform"
+              "qgnomeplatform-qt6"
+              "qadwaitadecorations"
+              "qadwaitadecorations-qt6"
+              [ "libsForQt5" "plasma-integration" ]
+              [ "libsForQt5" "qt5ct" ]
+              [ "libsForQt5" "qtstyleplugins" ]
+              [ "libsForQt5" "systemsettings" ]
+              [ "lxqt" "lxqt-config" ]
+              [ "lxqt" "lxqt-qtplugin" ]
+              [ "qt6Packages" "qt6ct" ]
+              [ "qt6Packages" "qt6gtk2" ]
+            ];
+            description = ''
+              Platform theme to use for Qt applications.
+
+              Some examples are
+
+              `gtk`
+              : Use GTK theme with
+                [`qtstyleplugins`](https://github.com/qt/qtstyleplugins)
+
+              `gtk3`
+              : Use [GTK3 integration](https://github.com/qt/qtbase/tree/dev/src/plugins/platformthemes/gtk3)
+                for file picker dialogs, font and theme configuration
+
+              `adwaita`
+              : Use Adwaita theme with
+                [`qadwaitadecorations`](https://github.com/FedoraQt/QGnomePlatform)
+
+              `gnome` (deprecated)
+              : Use GNOME theme with
+                [`qgnomeplatform`](https://github.com/FedoraQt/QGnomePlatform).
+                Is no longer maintained so prefer `adwaita`.
+
+              `lxqt`
+              : Use LXQt theme style set using the
+                [`lxqt-config-appearance`](https://github.com/lxqt/lxqt-config)
+                application
+
+              `qtct`
+              : Use Qt style set using
+                [`qt5ct`](https://github.com/desktop-app/qt5ct)
+                and [`qt6ct`](https://github.com/trialuser02/qt6ct)
+                applications
+
+              `kde`
+              : Use Qt settings from Plasma
+            '';
+          };
+          package = lib.mkOption {
+            type = with lib.types; nullOr (either package (listOf package));
+            default = null;
+            example =
+              lib.literalExpression "[pkgs.adwaita-qt pkgs.adwaita-qt6]";
+            description = ''
+              Theme package to be used in Qt5/Qt6 applications.
+              Auto-detected from {option}`qt.platformTheme.name` if possible.
+              See its documentation for available options.
+            '';
+          };
+        };
+      in lib.mkOption {
         type = with lib.types;
-          nullOr (enum [ "gtk" "gtk3" "gnome" "lxqt" "qtct" "kde" ]);
+          nullOr
+          (either (enum [ "gtk" "gtk3" "gnome" "adwaita" "lxqt" "qtct" "kde" ])
+            (lib.types.submodule { options = newOption; }));
         default = null;
```

Comment:
Maybe we should set this option to `internal` so it doesn't show in docs?

Reply:
No, I think it's better if users of the old option can still find it and then see the deprecation message.

Reply:
IMO, no because we don't want users to use those options. We want new users to see the new option only, and old users will receive the deprecation message.

Reply:
True, I'll set it to internal then. Though now `nix build .#docs-html` throws an error because [release-notes/rl-2105.md](https://github.com/nix-community/home-manager/blob/master/docs/release-notes/rl-2105.md?plain=1#L145) references the option. Because it is an old release note, we can probably remove the reference.

Another thing that I noticed is that the `.name` and `.package` submodule options don't get listed (even if `qt.platformTheme` isn't internal) in the reference documentation. The same happens with `services.syncthing.tray` which also has hidden submodules. Do you know of an option where this doesn't happen to see how we could solve this?

Reply:
Or we just accept it like that and keep `qt.platformTheme` not internal. I have a feeling the doc generator isn't capable of resolving submodules, especially because they're like normal [modules](https://discourse.nixos.org/t/override-submodule-options/12165). 

At least the correct option is listed in its description, and it says `or (submodule)` as type:
![image](https://github.com/nix-community/home-manager/assets/89387771/ebabfd96-3d1b-4a28-96fd-dedd6bf81764)
Not the most user-friendly, though.

Reply:
If this hides the submodule documentation themselves then I don't think setting it as internal is worth it. Sorry about the extra work.

Reply:
The submodules are hidden no matter the `internal` setting. I'll resolve this thread then.